Issue 8

tom-nslt opened the pull request.

jpwahle committed:
Add current activated env for mypy

jpwahle committed:
Remove unused requests deps

jpwahle committed:
Exclude .venv for performance reasons.

jpwahle committed:
Add isort to poetry poe command palette.

jpwahle committed:
flake8: ignore line break before binary operators

tom-nslt committed:
First draft of authentication middleware

tom-nslt committed:
Completed auth middleware

tom-nslt committed:
Improved auth documentation, Removed unnecessary check in authenticate_user

tom-nslt committed:
Improved documentation

tom-nslt committed:
Finalized version of Issue #8

tom-nslt committed:
Reenabled BLK100 checking and code style is now black #minor

tom-nslt committed:
Fixed minor issues + added sample env file #minor

tom-nslt committed:
Fixed minor issues #minor

jpwahle commented on nlp_land_prediction_endpoint/models/model_token_data.py:
```diff
@@ -0,0 +1,24 @@
+        isActive (Optional[bool]): flag indicating whether the user is still active
+    """
+
+    # XXX-TN change isAdmin to groups
```
I agree, we will solve this by groups later. To not make this an artifact in the code, can you please open an issue for that and link the line of code from that commit?

jpwahle commented on nlp_land_prediction_endpoint/models/model_token_data.py:
```diff
@@ -0,0 +1,24 @@
+    """
+
+    # XXX-TN change isAdmin to groups
+    # TODO-TN isActive is currently getting ignored
```
Can you please also open an issue for that and link this line of code of your commit in the issue?

jpwahle commented on nlp_land_prediction_endpoint/sample.env:
```diff
@@ -0,0 +1,6 @@
+AUTH_LOGIN_PROVIDER="http://127.0.0.1"
+AUTH_LOGIN_ROUTE="/api/v0/auth/login/service"
```
Ideally, we want this to be generic according to the release version.

jpwahle commented on tests/middleware/test_auth_route.py:
```diff
@@ -0,0 +1,174 @@
+        refresh_endpoint (str): Endpoint prefix.
+        dummy_token (str): a valid dummy token
+    """
+    response = client.post(refresh_endpoint, headers={"Authorization": f"Bearer {dummy_token}"})
```
Can we please validate that we retrieved a valid JWT token and that its expiration is longer than the dummy_token?

jpwahle commented on tests/middleware/test_auth_route.py:
```diff
@@ -0,0 +1,174 @@
+    os.environ["AUTH_LOGIN_PROVIDER"] = "http://127.0.0.1"
+    os.environ["AUTH_TOKEN_ROUTE"] = login_endpoint
+    response = client.post(login_endpoint, json=dummy_user.dict())
+    assert response.status_code == 200
```
Can we validate, that we get a valid JWT token?

jpwahle reviewed (commented):
I will approve the pull request when the minor comments are addressed. Should be easy to fix :)
Good Job!

tom-nslt committed:
Resolved the comments and also improved the structure a bit

https://github.com/ag-gipp/NLP-Land-prediction-endpoint/pull/12#discussion_r771201127
https://github.com/ag-gipp/NLP-Land-prediction-endpoint/pull/12#discussion_r771200411

tom-nslt committed:
Additional functions for more flexibilty; More .env options

tom-nslt committed:
Added function to request the backend version #minor

tom-nslt committed:
Fallback for missing env file #minor

jpwahle reviewed (approved):
Looks good to me!

jpelhaW commented:
@tom-nslt please merge when you feel comfortable!
